ci-operator/step-registry/cucushift/installer/check/gcp/user-tags/cucushift-installer-check-gcp-user-tags-commands.sh

```diff
@@ -0,0 +1,77 @@
+#!/bin/bash
+
+set -o nounset
+set -o errexit
+set -o pipefail
+
+if [ ! -f "${SHARED_DIR}/user_tags_sa.json" ]; then
+  echo "$(date -u --rfc-3339=seconds) - ERROR: Failed to find the key file of the IAM service-account for userTags testing on GCP."
+  exit 1
+fi
+
+if [ -f "${SHARED_DIR}/kubeconfig" ] ; then
+    export KUBECONFIG=${SHARED_DIR}/kubeconfig
+fi
+
+if [ -f "${SHARED_DIR}/proxy-conf.sh" ] ; then
+    source "${SHARED_DIR}/proxy-conf.sh"
+fi
+
+INFRA_ID="$(oc get infrastructures.config.openshift.io cluster -o jsonpath='{.status.infrastructureName}')"
+GCP_REGION="${LEASED_RESOURCE}"
+
+GOOGLE_PROJECT_ID="$(< ${CLUSTER_PROFILE_DIR}/openshift_gcp_project)"
+export GCP_SHARED_CREDENTIALS_FILE="${SHARED_DIR}/user_tags_sa.json"
+sa_email=$(jq -r .client_email ${GCP_SHARED_CREDENTIALS_FILE})
+if ! gcloud auth list | grep -E "\*\s+${sa_email}"
+then
+  gcloud auth activate-service-account --key-file="${GCP_SHARED_CREDENTIALS_FILE}"
+  gcloud config set project "${GOOGLE_PROJECT_ID}"
+fi
+
+# User-defined tags validation. It will check if each user-defined tag is applied. 
+# Return non-zero is one or more user-defined tag absent. 
+# $1 - the current tags of the resource under question
+function validate_user_tags() {
+  local -r current_tags="$1";  shift
+
+  printf '%s' "${USER_TAGS:-}" | while read -r PARENT KEY VALUE || [ -n "${PARENT}" ]
+  do
+    a_tag_value="${PARENT}/${KEY}/${VALUE}"
+    space_pattern="\ "
+    if [[ "${a_tag_value}" =~ ${space_pattern} ]] && [ ${#a_tag_value} -gt 64 ]; then
```

Comment:
What is the purpose of this if block?

Reply:
When there's space(s) in the tag value, and the total length of the tag's PARENT/KEY/VALUE exceeds 64 characters, the output of the gcloud command (line 65) will do text wrapping, see below example. In such case, the comparison of the tag value will ignore the words wrapped to the next line (i.e. "Xing Xiao" in below example). Any suggestion? Thanks! 

```
---
namespacedTagKey: XXXXXXXXXXXX/Su.Shi-Die_Lian_Hua-Chun_Jing
namespacedTagValue: XXXXXXXXXXXX/Su.Shi-Die_Lian_Hua-Chun_Jing/Hua Tui Can Hong Qing
  Xing Xiao
tagKey: tagKeys/281475274599629
tagKeyParentName: projects/1042363005003
tagValue: tagValues/281477497267603

Reply:
the text wrapping is happening only when there's space(s) in the tag value, if there is no white spaces, then no text wrapping?

Reply:
Yes, it's word wrapping.